CppSamples/EditData/EditGeodatabaseWithTransactions/EditGeodatabaseWithTransactions.cpp

```diff
@@ -0,0 +1,613 @@
+  if (isInTransaction)
+  {
+    setMessageText("Transaction started.");
+  }
+  else if (m_requireTransaction)
+  {
+    setMessageText("Tap Start to begin a transaction.");
+  }
+  else
+  {
+    setMessageText("Tap on the map to add a feature.");
+  }
+
+  emit stopEditingEnabledChanged();
+  emit startEditingEnabledChanged();
+}
+
+void EditGeodatabaseWithTransactions::setRequireTransaction(bool require)
+{
+  if (!m_geodatabase)
+  {
+    return;
+  }
+
+  if (!require && m_geodatabase->isInTransaction())
+  {
+    setMessageText("Stop editing to end the current transaction.");
+    return;
+  }
+
+  m_requireTransaction = require;
+  emit requireTransactionChanged();
+
+  // Update status message based on requirement
+  if (require)
+  {
+    setMessageText("Tap Start to begin a transaction.");
+  }
+  else
+  {
+    setMessageText("Tap on the map to add a feature.");
+  }
+
+  // Update control states
+  bool isInTransaction = m_geodatabase->isInTransaction();
+  m_startEditingEnabled = require && !isInTransaction;
+  m_stopEditingEnabled = require && isInTransaction;
+
+  emit startEditingEnabledChanged();
+  emit stopEditingEnabledChanged();
+}
+
+void EditGeodatabaseWithTransactions::beginTransaction()
+{
+  if (!m_geodatabase)
+  {
+    return;
+  }
+
+  // Begin transaction for the geodatabase if its not in transaction
+  if (!m_geodatabase->isInTransaction())
+  {
+    m_geodatabase->beginTransaction();
+  }
+}
+
+void EditGeodatabaseWithTransactions::stopTransaction()
+{
+  if (!m_geodatabase || !m_mapView)
+  {
+    return;
+  }
+
+  // Ask the user if they want to commit or rollback the transaction
+  emit commitDialogRequested();
+}
+
+void EditGeodatabaseWithTransactions::commitTransaction()
+{
+  if (!m_geodatabase)
+  {
+    return;
+  }
+
+  // Commit the transaction to store the edits (this will also end the transaction)
+  m_geodatabase->commitTransaction();
+  setMessageText("Edits committed to geodatabase.");
+}
+
+void EditGeodatabaseWithTransactions::rollbackTransaction()
+{
+  if (!m_geodatabase)
+  {
+    return;
+  }
+
+  // Rollback the transaction to discard the edits (this will also end the transaction)
+  m_geodatabase->rollbackTransaction();
+  setMessageText("Edits discarded.");
+}
+
+void EditGeodatabaseWithTransactions::cancelTransaction()
+{
+  // User canceled - keep transaction active
+  setMessageText("Transaction started.");
+}
+
+void EditGeodatabaseWithTransactions::updateFeatureTypesForTable(GeodatabaseFeatureTable *table)
+{
+  if (!table)
+  {
+    return;
+  }
+
+  m_currentFeatureTypes.clear();
+
+  QStringList featureTypeNames;
+  for (const FeatureType &featureType : table->featureTypes())
+  {
+    featureTypeNames.append(featureType.name());
+  }
+
+  featureTypeNames.sort(Qt::CaseInsensitive);
+  m_currentFeatureTypes = featureTypeNames;
+
+  emit currentFeatureTypesChanged();
+}
+
+void EditGeodatabaseWithTransactions::setSelectedTableName(const QString &tableName)
```

Comment:
Please use consistent style. I won't comment on each one.

```suggestion
void EditGeodatabaseWithTransactions::setSelectedTableName(const QString& tableName)
```

Reply:
used full formatting that should take care of this.